### ISSUE TITLE
Add link to newest version of a page from any page

### ISSUE THREAD

**Issue by lahabana** [kumahq/kuma-website#857]
### Description

Useful when you are on the wrong version

**Comment by lahabana** [comment]
With the Jekyll migration this works well when picking the dropdown. What would be perfect is if the "here" link also switched just the version instead of moving to `kuma.io/docs/latest`.

<img width="726" alt="image" src="https://user-images.githubusercontent.com/702355/198995554-e23e0109-8527-41dc-b9e9-ec62e825e6b8.png">


@fabianrbz do you think that's feasible?

**Comment by fabianrbz** [comment]
We definitely could, I'll work on this once we ship all the bug fixes

**Comment by fabianrbz** [comment]
@lahabana I'm currently working on this, and there is a minor issue when the page you are currently on is not available in the latest release. In that case, clicking on `Go here` will return a 404 which is expected and the only thing we can do is to add a new entry to the `redirects` file for every file that was moved/renamed.

FWIW, the same thing happens when you select a different version in the dropdown on the side nav.

**Comment by lahabana** [comment]
Yes I don't see this as a huge issue. Though I guess it breaks broken link checker?

**Comment by fabianrbz** [comment]
ok, so it ran, and it wasn't that [bad](https://github.com/kumahq/kuma-website/actions/runs/3418503127/jobs/5691162126)
I expected to have even more broken links, I'll run it locally just to confirm.
WDYT? Should we add those? I might need some help with some of them though.

**Comment by fabianrbz** [comment]
So, are these redirects correct?

* `https://deploy-preview-1148--kuma.netlify.app/docs/2.0.x/security/zone-ingress-auth` to `https://deploy-preview-1148--kuma.netlify.app/docs/2.0.x/security/zoneproxy-auth/`
* `https://deploy-preview-1148--kuma.netlify.app/docs/2.0.x/security/zoneegress-auth` to `https://deploy-preview-1148--kuma.netlify.app/docs/2.0.x/security/zoneproxy-auth/`
* the ones related to `prometheus`, it's been deprecated, so I'm not sure where should we redirect them to

**Comment by lahabana** [comment]
The 2 redirects you pointed are indeed missed on new pages.
Yes the problem is pages related to removed stuff. How will we go about it as there isn't and shouldn't be a page for it? Could we maybe mark this link to either avoid the crawling? Or have a different text if `latest` doesn't have this page?

**Comment by fabianrbz** [comment]
We could definitely exclude "dead" links from the check.
However, I do believe we should display something else in the banner so as to avoid bad UX -this won't prevent them from getting a 404 If they choose a different version using the dropdown-. We could simply drop the link, or show something else. Thoughts?

**Comment by lahabana** [comment]
> We could simply drop the link, or show something else.
Yes I think that's good enough

**Comment by lahabana** [comment]
Ok rethinking about this how about we keep the page when they are removed but we add a message about the feature being removed and what it's replaced by?

That way there are 2 options:

1. Page is moved --> Add a redirect
2. Page is removed (feature doesn't exist for example) --> The page should still exist but explain what this is replaced by (or why it was removed).

2 will be necessary anyway once we move to single source for all versions as pages don't really get removed anymore so we'd still need to add a conditional version block anyway isn't it?

**Comment by fabianrbz** [comment]
Not quite, so as far as I understand if a page that was removed, is not referenced in the navigation file of newer versions we don't generate them. cc @cloudjumpercat

I would drop the link if it is not available, fewer things to maintain/keep track of after they have been deprecated.

**Comment by lahabana** [comment]
Yes if that's the case SGTM!

**Comment by cloudjumpercat** [comment]
> Not quite, so as far as I understand if a page that was removed, is not referenced in the navigation file of newer versions we don't generate them. cc @cloudjumpercat

Yep, we don't reference removed pages in the nav file. If the page was removed because a feature was removed/deprecated, that's something we should mention in the release notes.

**Comment by fabianrbz** [comment]
@lahabana now that https://github.com/kumahq/kuma-website/pull/1148 has been merged, I think we can close this one.
